bootstrap_cfn/iam.py

```diff
@@ -18,18 +20,299 @@ def __init__(self, aws_profile_name, aws_region_name='eu-west-1'):
+
+        Args:
+            cert1(dictionary): Dictionary of certificate data,
+                with certs, chains and keys
+            cert2(dictionary): Dictionary of certificate data,
+                with certs, chains and keys
+
+        Returns:
+            are_equal: True if the certficate data are equal,
+            false otherwise
+        """
+
+        are_equal = False
+
+        start_text = "-----BEGIN CERTIFICATE-----"
+        end_text = "-----END CERTIFICATE-----"
+        certs_are_equal = self.compare_text_between_strings(
+                                                                 cert_data1.get("cert", None),
+                                                                 cert_data2.get("cert", None),
+                                                                 start_text,
+                                                                 end_text)
+        if not certs_are_equal:
+            logging
+        else:
+            start_text = "-----BEGIN CERTIFICATE-----"
+            end_text = "-----END CERTIFICATE-----"
+            chains_are_equal = self.compare_text_between_strings(
+                                                          cert_data1.get("chain", None),
+                                                          cert_data2.get("chain", None),
+                                                          start_text,
+                                                          end_text)
+            if not chains_are_equal:
+                logging.info("IAM::compare_certificate_data: "
+                             "Certificate chain data is not equal")
+            else:
+                are_equal = True
+
+        return are_equal
+
+    def compare_text_between_strings(self,
+                                     text1,
+                                     text2,
+                                     start_text,
+                                     end_text):
+        are_equal = False
+        if (text1 and text2 and (len(text1) > 0) and (len(text2) > 0)):
+            # Get the actual key data
+            body1 = (text1.split(start_text))[1].split(end_text)[0]
+            body2 = (text2.split(start_text))[1].split(end_text)[0]
+            if body1 and body2:
+                are_equal = (body1 == body2)
+
+        return are_equal
+
+    def upload_certificate(self, cert_name, stack_name, ssl_data, force=False):
+        """
+        Upload a certificate
+
+        Args:
+            cert_name(string): The name of the certificate entry to look up
+            stack_name(string): The name of the stack
+            ssl_data(dictionary): The configuration data for this certificate
+                entry
+
+        Returns:
+            success(bool): True if certificate is uploaded, False otherwise
+        """
+        success = False
+
+        cert_body = ssl_data['cert']
+        private_key = ssl_data['key']
+        try:
+            cert_chain = ssl_data['chain']
+        except:
```

Comment:
This should be `except KeyError` (always good to catch exceptions as specifically as possible.